py/dml/c_backend.py

```diff
@@ -2137,7 +2141,32 @@ def generate_object_vtables_array():
         items.append(pointer)
     init = '{%s}' % (', '.join(items),)
     add_variable_declaration(
-        'void * const _object_vtables[%d]' % (len(dml.globals.objects)),
+        f'void * const _object_vtables[{len(dml.globals.objects)}]', init)
+
+def generate_log_object_assoc_array():
+    items = []
+    for node in dml.globals.objects:
+        object_node = (node if node.objtype in {'device', 'bank',
+                                                'subdevice', 'port'}
+                       else node.object_parent)
+        if (object_node is not dml.globals.device
+            # Anonymous banks, OR banks never given a name due to compile
+            # errors
+            and not object_node.name is None):
```

Comment:
We don't need this precondition: Anonymous banks don't instantiate templates with shared methods (and it's OK to ICE if they do). And after a compile error, we can output whatever (and I don't understand how this could give `name is None` -- I think we have never-triggered assertions that `name is None` implies `objkind in {'field','bank'}`.

Reply:
> We don't need this precondition
We need _a_ precondition. Because this _will_ run into anonymous banks, and the code under this `if` _will_ ICE if so. If not this check, then it'd be a check if the bank instantiates any template with a shared method.
> Anonymous banks don't instantiate templates with shared methods
It's certainly possible thanks to in-each; but I don't disagree that it's such a niche case we can simply ICE if we run into it.
> (and I don't understand how this could give `name is None`)
Not sure myself. `1.4/errors/EAMBINH` and `1.4/errors/EIDENT` ICEd when I restricted this precondition to DML 1.2, and went away when I didn't.

Reply:
1. and 2. are WRONG!!!!!! This check is not only relevant for anonymous banks themselves, they are relevant for any object _under_ an anonymous bank! I don't feel comfortable restricting all such objects to never instantiate any template with a shared method, and there I don't see why that would be better anyway, since the check I already have in place is so simple and does the right thing. Like, do we want to go out of our way just to _spite_ DML 1.2 users? Not to mention it's a breaking change.

Reply:
> BTW, my python linter (ruff) says X is not None is preferred over not X is None.
Yup, that was a goof. I'll correct if once you've done your quick look.